components/availability/src/Availability.svelte

```diff
@@ -334,27 +347,58 @@
     return epochs;
   }
 
-  $: days = scaleTime()
-    .domain([startDay, endDay])
-    .ticks(timeDay)
-    .filter((timestamp: Date) => {
-      if (show_weekends) {
-        return true;
-      } else {
-        return (
-          timestamp.toString() !== timeSaturday(timestamp).toString() &&
-          timestamp.toString() !== timeSunday(timestamp).toString()
-        );
+  function generateDayRange({
+    startDay,
+    endDay,
+    reverse = false,
+  }: {
+    startDay: Date;
+    endDay: Date;
+    reverse?: boolean;
+  }) {
+    let range = scaleTime()
```

Comment:
Looks like this is the base case for this recursive function. Can we reorganize the conditional statements below to make sure this only gets called if we're not calling 
```
range = generateDayRange({ ... })
```